x-pack/plugins/ml/public/application/datavisualizer/data_comparison/data_drift_index_patterns_editor.tsx

```diff
@@ -0,0 +1,370 @@
+    };
+  }, [referenceIndexPattern, productionIndexPattern, timeField, dataViews]);
+  const createDataViewAndRedirectToDataComparisonPage = async (createAdHocDV = false) => {
+    // Create adhoc data view
+    const indicesName = `${referenceIndexPattern},${productionIndexPattern}`;
+
+    const timeFieldName =
+      Array.isArray(timeField) && timeField.length > 0 ? timeField[0].value : undefined;
+
+    let dataView;
+
+    if (!foundDataViewId) {
+      const modifiedDataViewName =
+        dataViewName === '' ? `${indicesName}-${timeFieldName}` : dataViewName;
+      if (canEditDataView && createAdHocDV === false) {
+        dataView = await dataViews.createAndSave({
+          title: indicesName,
+          name: modifiedDataViewName,
+          timeFieldName,
+        });
+      } else {
+        dataView = await dataViews.create({
+          title: indicesName,
+          name: modifiedDataViewName,
+          timeFieldName,
+        });
+      }
+    }
+    const dataViewId = foundDataViewId ?? dataView?.id;
+    const url = await locator.getUrl({
+      page: ML_PAGES.DATA_COMPARISON,
+      pageState: {
+        index: dataViewId,
+        reference: referenceIndexPattern,
+        production: productionIndexPattern,
+        timeFieldName,
+      },
+    });
+
+    await navigateToPath(url);
+  };
+
+  const firstSetOfSteps = [
+    {
+      title: 'Pick index pattern for reference data indices',
+      children: (
+        <EuiFlexItem grow={false}>
+          <DataViewEditor
+            key={'reference'}
+            label={
+              <FormattedMessage
+                id="xpack.ml.dataDrift.indexPatternsEditor.referenceData"
+                defaultMessage="Index pattern for reference data indices"
+              />
+            }
+            dataViewEditorService={referenceDataViewEditorService}
+            indexPattern={referenceIndexPattern}
+            setIndexPattern={setReferenceIndexPattern}
+          />
+        </EuiFlexItem>
+      ),
+    },
+    {
+      title: 'Pick index pattern for comparison data indices',
+      children: (
+        <EuiFlexItem grow={false}>
+          <DataViewEditor
+            key={'comparison'}
+            label={
+              <FormattedMessage
+                id="xpack.ml.dataDrift.indexPatternsEditor.comparisonData"
+                defaultMessage="Index pattern for comparison data indices"
+              />
+            }
+            dataViewEditorService={productionDataViewEditorService}
+            indexPattern={productionIndexPattern}
+            setIndexPattern={setProductionIndexPattern}
+          />
+        </EuiFlexItem>
+      ),
+    },
+    {
+      title: 'Additional settings',
+      children: (
+        <EuiFlexGroup direction="column">
+          {combinedTimeFieldOptions.length > 0 ? (
+            <EuiFormRow
+              label={i18n.translate('xpack.ml.dataDrift.indexPatternsEditor.timeField', {
+                defaultMessage: 'Time field',
```

Comment:
`Timestamp field` would be consistent with the Stack Management Data views page.